src/componentsNext/Pagination/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { Dispatch, SetStateAction } from "react";
+import {
+  PaginationLink,
+  Pagination as PaginationWrapper,
+} from "~/design/Pagination";
+
+import describePagination from "./describePagination";
+
+type SetState<T> = Dispatch<SetStateAction<T>>;
+
+export const Pagination = ({
+  itemsPerPage,
+  totalItems,
+  offset,
+  setOffset,
+}: {
+  itemsPerPage: number;
+  totalItems?: number;
+  offset: number;
+  setOffset: SetState<number>;
+}) => {
+  const setOffsetByPageNumber = (pageNumber: number) =>
+    (pageNumber - 1) * itemsPerPage;
+
+  const numberOfItems = totalItems ?? 0;
+  const pages = Math.ceil(numberOfItems / itemsPerPage);
+  const currentPage = Math.ceil(offset / itemsPerPage) + 1;
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === pages;
+
+  const previousPage = currentPage > 1 ? currentPage - 1 : null;
+  const nextPage = currentPage < pages ? currentPage + 1 : null;
+
+  const paginationDescription = describePagination({ currentPage, pages });
+
+  return (
+    <PaginationWrapper>
+      <PaginationLink
+        icon="left"
+        onClick={() =>
+          previousPage && setOffset(setOffsetByPageNumber(previousPage))
+        }
+        disabled={isFirstPage}
+      />
+      {paginationDescription.map((page, index) => {
+        if (page === "…")
+          return (
+            <PaginationLink key={index} disabled>
+              ...
```

Comment:
```suggestion
              …
```

Reply:
Damn. I thought this was a change that was 100 % covered by typescript an i just had to update all the places where TS was screaming at me. But I was wrong. Should we use {page} instead?

```typescript
 if (page === "…")
   return (
     <PaginationLink key={index} disabled>
       {page}
```

Reply:
Yes, it makes a lot of sense to use `{page}`. 

Actually, that makes me wonder if it would be better to not to have an if-block for the ellipsis at all. This works (but note the added complexity in the onClick statement necessary to avoid TS errors)

```tsx
// remove the if-block for if (page === "…"), then below that:
        const isActive = currentPage === page;
        const isEllipsis = page === "…";
        return (
          <PaginationLink
            key={index}
            active={isActive}
            onClick={() => {
              !isEllipsis &&
                !isActive &&
                setOffset(setOffsetByPageNumber(page));
            }}
            disabled={isEllipsis}
          >
            {page}
          </PaginationLink>

Reply:
Good idea, that is much cleaner!